packages/wallet/src/services/DelegationTracker/RewardsHistory.ts

```diff
@@ -34,11 +34,20 @@ export const createRewardsHistoryProvider =
 
 export type RewardsHistoryProvider = ReturnType<typeof createRewardsHistoryProvider>;
 
-const firstDelegationEpoch$ = (transactions$: Observable<TxWithEpoch[]>) =>
+const firstDelegationEpoch$ = (transactions$: Observable<TxWithEpoch[]>, rewardAccounts: Cardano.RewardAccount[]) =>
   transactions$.pipe(
     map((transactions) =>
       first(
-        transactions.filter(({ tx }) => transactionHasAnyCertificate(tx, [Cardano.CertificateType.StakeDelegation]))
+        transactions.filter(({ tx }) => {
```

Comment:
Let's map transactions to either 'null' or `{epoch}` here instead. This function has to return epoch of the first tx with delegation from own reward account.

```suggestion
        transactions.map(({ tx }) => {
```

Reply:
Do you mean like this?
```ts
  transactions$.pipe(
    map((transactions) =>
      first(
        transactions.map(({ tx, epoch }) => {
          const inspectTx = createTxInspector({
            signedCertificates: signedCertificatesInspector(rewardAccounts, [Cardano.CertificateType.StakeDelegation])
          });
          return inspectTx(tx).signedCertificates.length > 0 ? { epoch } : null;
        })
      )
    ),
    map((tx) => (util.isNotNil(tx) ? tx.epoch + 3 : null)),
    distinctUntilChanged()
```
with this if the first transaction in `transactions$` is not signed or is not a delegation it'll return null, before it used to filter those out and get the first transaction that was a delegation and was signed

Reply:
Ah yes, you need to do `map(...).filter(util.isNotNil) then`